pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/batching/AbstractBatchedMetadataStore.java

```diff
@@ -99,7 +103,13 @@ public void close() throws Exception {
     private void flush() {
         while (!readOps.isEmpty()) {
             List<MetadataOp> ops = new ArrayList<>();
-            readOps.drain(ops::add, maxOperations);
+            for (int i = 0; i < maxOperations; i++) {
```

Comment:
This one should be ok, since it's already done in a loop: `while (!readOps.isEmpty()) {...}`